### PR TITLE
Ingest structured ES deprecation logs

### PR DESCRIPTION
**Pull request by ycombinator**
This PR teaches the `elasticsearch/deprecation` fileset to ingest structured Elasticsearch deprecation logs.

This PR takes the same approach as https://github.com/elastic/beats/pull/10352, in that it creates an entrypoint pipeline, `pipeline.json`, that delegates further processing of a log entry depending on what it sees as the first character of the entry:
- If the first character is `{`, it assumes the log line is structured as JSON and delegates further processing to `pipeline-json.json`.
- Else, it assumes the log line is plaintext and delegates further processing to `pipeline-plaintext.json`.

### CODE REVIEW

File: filebeat/module/elasticsearch/deprecation/test/elasticsearch_deprecation.log-expected.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     {
         "@timestamp": "2018-04-23T16:40:13.737Z",
         "ecs.version": "1.0.0-beta2",
-        "elasticsearch.server.component": "o.e.d.a.a.i.t.p.PutIndexTemplateRequest",
+        "elasticsearch.component": "o.e.d.a.a.i.t.p.PutIndexTemplateRequest",
```
Comment (ruflin) [inline filebeat/module/elasticsearch/deprecation/test/elasticsearch_deprecation.log-expected.json]: Breaking change or never released?

Comment (ycombinator) [reply filebeat/module/elasticsearch/deprecation/test/elasticsearch_deprecation.log-expected.json]: Breaking change. Felt odd to me that the `deprecation` fileset was writing into  the `server` fileset's namespace. So instead I decided to promote the `component` field up to their common ancestor.

I'll add an additional CHANGELOG entry for this breaking change.

**Comment by ycombinator** [comment]
@ruflin Rebased and addressed review feedback. Ready for another pass, when you have a chance. Thanks!

**Comment by ycombinator** [comment]
jenkins, test this

**Pull request merged**